web/packages/teleport/src/SessionRecordings/view/player/tty/decoding.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * Teleport
+ * Copyright (C) 2025 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {
+  EventType,
+  type SerializedTerminal,
+  type TerminalSize,
+  type TtyEvent,
+} from './types';
+
+const responseHeaderSize = 17;
+
+export function decodeTtyEvent(buffer: ArrayBuffer): TtyEvent {
```

Comment:
It would be great to provide a link to the Go code that encodes these events to make it easy to figure out where the other side of the pipeline is.

Reply:
Will do once that PR has been merged